### ISSUE TITLE
WriteError lmdb error code -30792

### ISSUE THREAD

**Issue by TianQingX** [man-group/ArcticDB#440]
### Describe the bug

Trying to write data into LMDB but fail on windows

### Steps/Code to Reproduce

```python
import os
import pandas as pd
from tqdm import tqdm
from arcticdb import Arctic

def read_data(path):
    # Use read_csv function to read data, convert datetime string to datetime type, and set it as the index of DataFrame
    df = pd.read_csv(path, parse_dates=['datetime'], index_col='datetime', engine = "pyarrow")
    # Convert column names to start with an uppercase letter
    df = df.rename(columns=lambda x: x.capitalize())
    df = df[['Open', 'High', 'Low', 'Close', 'Amount']]

    # Extract filename as stock ID
    base = os.path.basename(path)
    stock_id = os.path.splitext(base)[0]  # Remove file extension to get stock ID

    # Add the stock ID to the DataFrame
    df["id"] = stock_id

    return df, stock_id

def process_file(file_path, library):
    df, stock_id = read_data(file_path)
    try:
        # Try to update
        library.update(stock_id, df)
        print(f"Successfully updated {stock_id}")
    except:
        # If update fails, try to write
        print(f"Failed to update, trying to write {stock_id}")
        library.write(stock_id, df)

def process_directory(directory_path, library):
    # Get all files in the directory
    files = os.listdir(directory_path)
    csv_files = [f for f in files if f.endswith('.csv')]

    # Use tqdm to create a progress bar
    for csv_file in tqdm(csv_files):
        # Call process_file function for each file
        process_file(os.path.join(directory_path, csv_file), library)

# Process all directories within a specified range of years
ac = Arctic('lmdb:///ArcticDBStore')
aStock = ac.get_library('AStock')
start_year = 2005
end_year = 2020
for year in range(start_year, end_year + 1):
    directory_path = os.path.join(r"C:\backup\1min_csv", str(year))
    process_directory(directory_path, aStock)
```

### Expected Results

Write into LMDB

### OS, Python Version and ArcticDB Version

Python: 3.8.16 (default, Mar  2 2023, 03:18:16) [MSC v.1916 64 bit (AMD64)]
OS: Windows-10-10.0.19041-SP0
ArcticDB: 1.2.1

### Backend storage used

LMDB

### Additional Context

```
(trade) C:\Users\ps>python C:\backup\ML\Ngboost\ArcticDBfor1minBars.py
  0%|                                                                                         | 0/1365 [00:00<?, ?it/s]try to write sh600000
  0%|                                                                                         | 0/1365 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "C:\backup\ML\Ngboost\ArcticDBfor1minBars.py", line 26, in process_file
    library.update(stock_id, df)
  File "C:\Users\ps\anaconda3\envs\trade\lib\site-packages\arcticdb\version_store\library.py", line 723, in update
    return self._nvs.update(
  File "C:\Users\ps\anaconda3\envs\trade\lib\site-packages\arcticdb\version_store\_store.py", line 790, in update
    vit = self.version_store.update(
arcticdb_ext.exceptions.InternalException: (Invalid lmdb error code -30792 while putting key d:sh600000:1:0x9fe04e1c9c511e6a@1685422629438385700[1104831000000000000,1135954740000000001])

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\backup\ML\Ngboost\ArcticDBfor1minBars.py", line 50, in <module>
    process_directory(directory_path, aStock)
  File "C:\backup\ML\Ngboost\ArcticDBfor1minBars.py", line 41, in process_directory
    process_file(os.path.join(directory_path, csv_file), library)
  File "C:\backup\ML\Ngboost\ArcticDBfor1minBars.py", line 31, in process_file
    library.write(stock_id, df)
  File "C:\Users\ps\anaconda3\envs\trade\lib\site-packages\arcticdb\version_store\library.py", line 373, in write
    return self._nvs.write(
  File "C:\Users\ps\anaconda3\envs\trade\lib\site-packages\arcticdb\version_store\_store.py", line 570, in write
    vit = self.version_store.write_versioned_dataframe(
arcticdb_ext.exceptions.InternalException: (Invalid lmdb error code -30792 while putting key d:sh600000:1:0x9fe04e1c9c511e6a@1685422629521433400[1104831000000000000,1135954740000000001])
```

**Comment by jjerphan** [comment]
Hi @TianQingX, thank you for reporting this bug.

We do not have access to the data you are using. Can you provide a minimal reproducible example with synthetic data?

**Comment by qc00** [comment]
This is most likely another manifestation of #229. In short, the maximum amount of data that can be written to a library in Windows is 128MB at the moment.

**Comment by mehertz** [comment]
We've clearly gotta fix #229...

**Comment by jjerphan** [comment]
Hi @TianQingX, have you seen my previous message (i.e. https://github.com/man-group/ArcticDB/issues/440#issuecomment-1567959095)?
